## Практическое задание № 6 ##

### Урок 6.  Взаимосвязь величин. Параметрические и непараметрические показатели корреляции. Корреляционный анализ.


1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

Выберите тему для проектной работы по курсу Теории вероятностей и математической статистики
и напишите ее в комментарии к Практическому заданию

In [1]:
import math
import numpy as np

#### Задача N 1
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [4]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

$$cov_{XY} = M(XY) - M(X)M(Y)$$

In [ ]:
cov = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
cov

In [13]:
np.cov(zp, ks, ddof=0)[0, 1]

9157.84

> значения 9157.84 ковариации равны

$$r_{XY} = \frac{cov_{XY}}{\sigma_{X}\sigma_{Y}}$$

In [18]:
r1 = cov / (np.std(zp, ddof=0) * np.std(ks, ddof=0))
r1

0.8874900920739158

In [20]:
r2 = np.corrcoef(zp, ks)[0, 1]
r2

0.8874900920739162

> значения 0.8875 коэффициента ковариации равны

#### Задача N 2
Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

$$T_{1,2} = \overline{X} \pm t_{\alpha/2}\cdot\frac{s_0}{\sqrt{n}} $$

In [27]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
iq.sort()
iq

array([ 99, 107, 111, 115, 115, 122, 125, 125, 131, 131])

In [23]:
n = len(iq)
n

10

In [31]:
t2a = 2.262
t1 = iq.mean() - (t2a * iq.std(ddof=1) / np.sqrt(n))
t2 = iq.mean() + (t2a * iq.std(ddof=1) / np.sqrt(n))
(t1, t2)

(110.55660776308164, 125.64339223691834)

> Доверительный интервал равен [110.6, 125.6]  с надежностью 0.95.

#### Задача N 3
Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

$$T_{1,2} = \overline{X} \pm z_{\alpha/2}\cdot\frac{s_0}{\sqrt{n}} $$

In [32]:
t1 = 174.2 - (1.96 * 5 / np.sqrt(27))
t2 = 174.2 + (1.96 * 5 / np.sqrt(27))
(t1, t2)

(172.31398912064722, 176.08601087935276)

> Доверительный интервал равен [172.3, 176.1] с надежностью 0.95.

